src/Microsoft.AspNetCore.StaticFiles/ResponseCacheFilter.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.StaticFiles
+{
+    internal class ResponseCacheFilter : IResponseCacheFilter
+    {
+        /// <summary>
+        /// Apply the given cacheProfile to the <see cref="HttpContext"/>
+        /// </summary>
+        /// <param name="context">The current <see cref="HttpContext"/></param>
+        /// <param name="cacheProfile">The profile which contains the cache control settings</param>
+        public void ApplyCacheProfile(HttpContext context, CacheProfile cacheProfile)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+            if (cacheProfile == null)
+            {
+                throw new ArgumentNullException(nameof(cacheProfile));
+            }
+
+            var noStore = cacheProfile.NoStore ?? false;
+            var varyByHeader = cacheProfile.VaryByHeader;
+            var location = cacheProfile.Location ?? ResponseCacheLocation.Any;
+
+            if (!noStore)
+            {
+                // Duration MUST be set unless NoStore is true.
+                if (cacheProfile.Duration == null)
+                {
+                    throw new InvalidOperationException(
+                            Resources.FormatResponseCache_SpecifyDuration(
+                                nameof(CacheProfile.NoStore), nameof(CacheProfile.Duration)));
+                }
+            }
+
+            var headers = context.Response.Headers;
+
+            // Clear all headers
+            headers.Remove(HeaderNames.Vary);
+            headers.Remove(HeaderNames.CacheControl);
+            headers.Remove(HeaderNames.Pragma);
```

Comment:
I guess any custom middleware _could_ set them, though it's obviously unlikely at this stage. Clearing them just ensures that if they have been set previously, but are unset in the CacheProfile, you aren't left with mixed configuration - the `CacheProfile` settings completely override any previous values..